### PR TITLE
feat: filter storage mode responses by submission id

### PR DESCRIPTION
**Pull request by shuli-ogp**
Fixes a bug with #174 for release. #174 was reverted in #358.

### CODE REVIEW

**Review by liangyuanruo — APPROVED** [review]
lgtm - could i trouble you to test this in staging to guarantee performance before merging, so that it definitely makes the next release? it's been languishing awhile.

FYI @frankchn, there was a bug with the conditional logic. Return statements are booby traps when used in a callback pattern.

File: src/app/controllers/encrypt-submissions.server.controller.js
Comment (mantariksh) [inline src/app/controllers/encrypt-submissions.server.controller.js:244]: This should follow our standard logging format (`message`, `meta`) to make searching the logs easier

**Review by mantariksh — COMMENTED** [review]
minor comment on the logging format. I also think it might be worth the effort to extract some of this function into a service, or at least use `async`/`await` to make the control flow a little more obvious, but will defer to @liangyuanruo

**Comment by mantariksh** [comment]
fixed the log statement and tested on staging-alt, nothing unusual seems to be happening with the staging-alt db so I think we're good to go

**Pull request merged**
